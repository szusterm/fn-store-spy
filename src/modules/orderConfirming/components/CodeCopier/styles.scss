@import '~material-colors/dist/colors';
@import '~scss/_opacity-colors';
@import '~scss/_transition';

.code-copier {
	width: 100%;
	padding: 15px;

	&--code {
		cursor: pointer;

		padding: 8px;
		font-size: 3.5em;
		font-weight: 700;
		letter-spacing: 0.1em;
		text-align: center;
		text-transform: uppercase;

		background-color: $md-blue-500;
		border-radius: 15px;

		@include white-color(0.95);
		@include transition(sh);

		&:hover, &:focus {
			background-color: $md-blue-600;

			@include white-color(1);
		}

		&::selection {
			@include white-background(0.9);
			color: $md-blue-400;
		}

		&--copied {
			background-color: $md-green-500;

			&::selection {
				color: $md-green-400;
			}

			&:hover, &:focus {
				background-color: $md-green-600;
			}
		}
	}

	&--info {
		padding: 8px;
		text-align: center;
		text-transform: uppercase;
		font-weight: 700;

		.info {
			&--how-to {
				@include black-color(0.3);
			}

			&--copied {
				color: $md-green-300;
			}
		}
	}
}