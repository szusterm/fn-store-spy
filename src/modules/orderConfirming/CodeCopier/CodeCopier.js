import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import './styles.scss';

export class CodeCopier extends Component {
	render() {
		return (
			<input
				className={'code-copier'}
			/>
		);
	}
}

CodeCopier.propTypes = {
	code: PropTypes.string //redux
};

const mapStateToProps = (state) => {
	const {code} = state.orderConfirming;

	return {code};
};

export default connect(mapStateToProps, null)(CodeCopier);