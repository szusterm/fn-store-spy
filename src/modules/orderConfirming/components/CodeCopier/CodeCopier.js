import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {CopyToClipBoard} from 'react-copy-to-clipboard';

import './styles.scss';

export class CodeCopier extends Component {
	render() {
		const {code} = this.props;

		return (
			<input
				className={'code-copier'}
				readOnly={true}
				value={code}
			/>
		);
	}
}

CodeCopier.propTypes = {
	code: PropTypes.string //redux
};

const mapStateToProps = (state) => {
	const {code} = state.orderConfirming;

	return {code};
};

export default connect(mapStateToProps, null)(CodeCopier);