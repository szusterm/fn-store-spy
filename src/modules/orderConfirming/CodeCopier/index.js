import CodeCopier from './CodeCopier';

export default CodeCopier;