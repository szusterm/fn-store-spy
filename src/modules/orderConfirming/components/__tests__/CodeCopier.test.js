import React from 'react';
import {shallow} from 'enzyme';
import {CopyToClipboard} from 'react-copy-to-clipboard';

import {CodeCopier} from '../CodeCopier/CodeCopier';

const setup = (propOverrides) => {
	const props = Object.assign({
		code: 'fnh4y' //redux
	}, propOverrides);

	const wrapper = shallow(<CodeCopier {...props}/>);

	return {props, wrapper};
};

describe('CodeCopier Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('shows the code from a store', () => {
		const {wrapper, props} = setup({
			code: 'fn5g6u'
		});

		const inputValue = wrapper.find('.code-copier--code').text();

		expect(inputValue).toBe(props.code);
	});

	it('copies the code after click through CopyToClipboard', () => {
		const {wrapper, props} = setup({
			code: 'fn5g6u'
		});

		const textToCopy = wrapper.find(CopyToClipboard).props().text;

		expect(textToCopy).toBe(props.code);
	});

	it('changes information text after copy', () => {
		const {wrapper} = setup();

		const howToTextClass = '.info--how-to';
		const copiedTextClass = '.info--copied';

		expect(wrapper.find(howToTextClass).exists()).toBe(true);
		expect(wrapper.find(copiedTextClass).exists()).toBe(false);

		wrapper.setState({copied: true});

		expect(wrapper.find(howToTextClass).exists()).toBe(false);
		expect(wrapper.find(copiedTextClass).exists()).toBe(true);
	});

	it('changes information text after copy', () => {
		const {wrapper} = setup();

		const codeCopiedClass = '.code-copier--code--copied';

		expect(wrapper.find(codeCopiedClass).exists()).toBe(false);

		wrapper.setState({copied: true});

		expect(wrapper.find(codeCopiedClass).exists()).toBe(true);
	});
});