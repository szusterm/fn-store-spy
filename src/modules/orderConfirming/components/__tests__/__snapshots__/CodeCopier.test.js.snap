// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeCopier Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CodeCopier
    code="fnh4y"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="code-copier"
      >
        fnh4y
      </div>,
      "onCopy": undefined,
      "options": undefined,
      "text": "fnh4y",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "fnh4y",
        "className": "code-copier",
      },
      "ref": null,
      "rendered": "fnh4y",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="code-copier"
        >
          fnh4y
        </div>,
        "onCopy": undefined,
        "options": undefined,
        "text": "fnh4y",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "fnh4y",
          "className": "code-copier",
        },
        "ref": null,
        "rendered": "fnh4y",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
